operators/to_mapbox/__init__.py

```diff
@@ -0,0 +1,152 @@
+import dataflows as DF
+from dataflows.helpers.resource_matcher import ResourceMatcher
+from dataflows_airtable import load_from_airtable
+from srm_tools.logger import logger
+from ..mapbox_upload import upload_tileset
+from conf import settings
+
+
+def filter_airtable_dummies():
+    return DF.filter_rows(lambda r: not r['id'] == 'dummy')
+
+
+def unwind(from_key, to_key, key_transformer=None, resources=None):
+
+    # TODO - errors if to_key does not exist - using add_field below.
+    # TODO - from_key and to_key as equal length lists? like zip? See use case below with branch_name and organization_name.
+    # TODO - see if this is generic enough for the DF standard lib.
+    # TODO - always drop from_key, or, optionally drop?
+
+    def _unwinder(rows):
+        for row in rows:
+            try:
+                iter(row[from_key])
+                for value in row[from_key]:
+                    row[to_key] = value if key_transformer is None else key_transformer(value)
```

Comment:
I think you need to create a copy of row before yielding here - otherwise you're just going to keep changing and yielding the same object over and over again - so all of its copies are going to end up with the same value.

Reply:
Actually, that is not what I am seeing - this does unwind correctly, e.g.:

input:
```
name,organizations,response_id
הוסטל לאנשים עם מוגבלות,"[""recZm43eLrgBvbTZK"", ""recsA5yaphi07KUM9"", ""rec9zoJb5Mb3MYeQk"", ""recdjoVxDY1Zw9us0"", ""rec9qgaPXzclTfdNg"", ""recfTcFJNqEpOCCeQ"", ""rec2oZwyKkWw0Q0vV"", ""recjuGtTFvdoqkUix"", ""rec4YZtpZhCmDerr2"", ""recZic4vuKC8SmHVn"", ""recFKgxF1nqEBCtSw"", ""recF31yOEh48dwAxT"", ""recER1UhSwJH9PRYy"", ""recDrOCPfccibcMhK"", ""recHXdv0k0hlj0ciu"", ""recwgwxWAntlHALDG"", ""rec7MhfyQKKWIiFEh"", ""reckupOox2Lt4Myww"", ""recNy4qAugT9AONUb"", ""rec66n7VhEkVzEHlT"", ""recdpStaz0QnNbc3N"", ""recHbIdjfpnQvubw0"", ""recq3X5VIZ7YNkulZ"", ""recKUSlXBcvT5uqq5"", ""recqLg186d1gTVlmw"", ""recwyVl1aC1pV7ZL0"", ""reclrZTvhlfp7SrUo"", ""recAPUJZSSEVzE0eQ"", ""recOP2vkHIWYNNRob"", ""recmNxXtDTQlXQ6Rp"", ""recsr8CdKZoNWL7TA"", ""recTR7VNISqphzuV0"", ""recl6FWO8IiMFk37C"", ""recooY9DvWntHRKUE"", ""recwVRMT8Bu2hpbbI"", ""recj745PDjOKho2lt"", ""recUen4u42gaNzOii""]",human_services:care:residential_care

```

```
...
unwind('organizations', 'organization_key', resources=['service_response_lookup']),
...
```

produces:

```
name,organization_key,response_id
הוסטל לאנשים עם מוגבלות,recZm43eLrgBvbTZK,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recsA5yaphi07KUM9,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,rec9zoJb5Mb3MYeQk,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recdjoVxDY1Zw9us0,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,rec9qgaPXzclTfdNg,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recfTcFJNqEpOCCeQ,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,rec2oZwyKkWw0Q0vV,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recjuGtTFvdoqkUix,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,rec4YZtpZhCmDerr2,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recZic4vuKC8SmHVn,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recFKgxF1nqEBCtSw,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recF31yOEh48dwAxT,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recER1UhSwJH9PRYy,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recDrOCPfccibcMhK,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recHXdv0k0hlj0ciu,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recwgwxWAntlHALDG,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,rec7MhfyQKKWIiFEh,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,reckupOox2Lt4Myww,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recNy4qAugT9AONUb,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,rec66n7VhEkVzEHlT,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recdpStaz0QnNbc3N,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recHbIdjfpnQvubw0,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recq3X5VIZ7YNkulZ,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recKUSlXBcvT5uqq5,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recqLg186d1gTVlmw,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recwyVl1aC1pV7ZL0,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,reclrZTvhlfp7SrUo,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recAPUJZSSEVzE0eQ,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recOP2vkHIWYNNRob,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recmNxXtDTQlXQ6Rp,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recsr8CdKZoNWL7TA,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recTR7VNISqphzuV0,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recl6FWO8IiMFk37C,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recooY9DvWntHRKUE,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recwVRMT8Bu2hpbbI,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recj745PDjOKho2lt,human_services:care:residential_care
הוסטל לאנשים עם מוגבלות,recUen4u42gaNzOii,human_services:care:residential_care
```

Reply:
This depends on the processors afterwards - if they process the data row by row then you're in the clear.
if they buffer the rows (e.g. dump to sql), though, you might have a problem as they essentially would have the same object duplicated.

Reply:
Yes I saw that later, and fixed in the latest commit.